common/browser_context.go

```diff
@@ -451,44 +451,35 @@ func (b *BrowserContext) AddCookies(cookies goja.Value) {
 func (b *BrowserContext) addCookies(cookies goja.Value) error {
 	var cookieParams []network.CookieParam
 	if !gojaValueExists(cookies) {
-		return Error("cookies value is not set")
+		return Error("cookies argument must be set")
 	}
-
-	rt := b.vu.Runtime()
-	err := rt.ExportTo(cookies, &cookieParams)
-	if err != nil {
-		return fmt.Errorf("unable to export cookies value to cookieParams. %w", err)
+	if err := b.vu.Runtime().ExportTo(cookies, &cookieParams); err != nil {
+		return fmt.Errorf("cannot recognize cookie values: %w", err)
 	}
 
-	// Create new array of pointers to items in cookieParams
-	var cookieParamsPointers []*network.CookieParam
-	for i := 0; i < len(cookieParams); i++ {
-		cookieParam := cookieParams[i]
+	coookiesToSet := make([]*network.CookieParam, len(cookieParams))
+	for i, cookie := range cookieParams {
+		c := cookie
 
-		if cookieParam.Name == "" {
-			return fmt.Errorf("cookie name is not set. %#v", cookieParam)
+		if c.Name == "" {
+			return fmt.Errorf("cookie name must be set: %#v", c)
 		}
-
-		if cookieParam.Value == "" {
-			return fmt.Errorf("cookie value is not set. %#v", cookieParam)
+		if c.Value == "" {
+			return fmt.Errorf("cookie value must be set: %#v", c)
 		}
-
 		// if URL is not set, both Domain and Path must be provided
-		if cookieParam.URL == "" {
-			if cookieParam.Domain == "" || cookieParam.Path == "" {
-				return fmt.Errorf(
-					"if cookie url is not provided, both domain and path must be specified. %#v",
-					cookieParam,
-				)
-			}
+		if c.URL == "" && (c.Domain == "" || c.Path == "") {
+			const msg = "if cookie URL is not provided, both domain and path must be specified: %#v"
```

Comment:
Is this an optimisation to help the compiler? I think just placing the string in the `Errorf` should suffice still like you have in the errors above, have i missed something?

Reply:
Just a personal style thing :)